### PR TITLE
Fix: Not honoring gp3 volume iops and throughput values

### PR DESCRIPTION
**Pull request by vgunapati**
Fixes https://github.com/keikoproj/instance-manager/issues/379

### CODE REVIEW

File: controllers/providers/aws/constructors.go [hunk 1]
```diff
@@ -40,20 +38,22 @@ func (w *AwsWorker) GetAutoScalingBasicBlockDevice(name, volType, snapshot strin
 	if encrypt != nil {
 		device.Ebs.Encrypted = encrypt
 	}
-	if iops != 0 && strings.EqualFold(volType, "io1") {
+	if iops != 0 && common.ContainsEqualFold(AllowedVolumeTypesWithProvisionedIOPS, volType) {
 		device.Ebs.Iops = aws.Int64(iops)
 	}
+	if throughput != 0 && common.ContainsEqualFold(AllowedVolumeTypesWithProvisionedThroughput, volType) {
```
Comment (eytan-avisror) [inline controllers/providers/aws/constructors.go:44]: Is there a default on the SDK side if it's set to 0?

**Review by eytan-avisror — APPROVED** [review]
LGTM 
We can do a release if you need it and the functional-test is passing

File: controllers/providers/aws/constructors.go (hunk 1 above)
Comment (vgunapati) [reply controllers/providers/aws/constructors.go:44]: old SDK did not even have the option to configure Throughput, So I have to bump up to the new version, Tested `0` on gp3 for ProvisionedThroughput it was treated as `nil`, and the default value is set in both the launch template and launch configuration

Comment (eytan-avisror) [reply controllers/providers/aws/constructors.go:44]: 👌🏻

**Comment by vgunapati** [comment]
Looks like [functional-test](https://github.com/keikoproj/instance-manager/actions/runs/2895189866) is broken for a while, Do we have a doc on how to set this up if the cluster is deleted

**Comment by eytan-avisror** [comment]
No doc, but it pretty much needs a vanilla EKS cluster + upgrade manager

**Comment by vgunapati** [comment]
I don't have access to the settings page to update Access keys, I will follow up with the team

**Pull request merged**